Added custom name to capacity provider

carlolucadei opened the pull request:
## what

We added the way to customize the capacity provider name

## why

With the current change the default behaviour is preserved.
We applied this in order to avoid issues while creating a new capacity provider for another environment since for AWS it already exists due to a previous deploy ie in test env.

gberenice commented on ec2.tf:
```diff
@@ -103,7 +103,7 @@ module "autoscale_group" {
 resource "aws_ecs_capacity_provider" "ec2" {
   for_each = local.ec2_capacity_providers
-  name     = each.key
+  name     = each.value["name"] != null && each.value["name"] != "" ? each.value["name"] : each.key
```
```suggestion
  name     = coalesce(each.value["name"], each.key)
```
We can simplify this by using the `coalesce` function, which returns the first non-null, non-empty string from its arguments

carlolucadei replied:
@gberenice thanks for the tip

gberenice commented:
@carlolucadei can I also ask you to address this bats issue for `examples/complete`:
```
Test: check if terraform inputs have descriptions
File: input-descriptions.bats
---------------------------------
availability_zones is missing a description
region is missing a description
---------------------------------
```
Thanks in advance!

carlolucadei commented:
@gberenice I saw your last message, it could take a while to fix it

gberenice commented:
Ah, sorry, I should have been more explicit about what to do. Tests are fine, but the `bats` check was complaining about missing required descriptions for the inputs in the example.
Please update the file [examples/complete/variables.tf](https://github.com/cloudposse/terraform-aws-ecs-cluster/blob/main/examples/complete/variables.tf) to this content:
```hcl
variable "region" {
  type        = string
  description = "AWS region"
}

variable "availability_zones" {
  type        = list(string)
  description = "Availability zone IDs"
}
```

carlolucadei commented:
> Ah, sorry, I should have been more explicit about what to do. Tests are fine, but the `bats` check was complaining about missing required descriptions for the inputs in the example. Please update the file [examples/complete/variables.tf](https://github.com/cloudposse/terraform-aws-ecs-cluster/blob/main/examples/complete/variables.tf) to this content:
> 
> ```hcl
> variable "region" {
>   type        = string
>   description = "AWS region"
> }
> 
> variable "availability_zones" {
>   type        = list(string)
>   description = "Availability zone IDs"
> }
> ```

@gberenice I got it. Thanks! Maybe it will be required to perform squash commit...
PS: How can I perform bats check? (sorry I'm a newby)

gberenice commented:
/terratest

gberenice commented:
@carlolucadei the last thing is left - please run `terraform fmt` for `examples/complete`

gberenice commented:
/terratest

carlolucadei commented:
> /terratest

I'm sorry, @gberenice  it’s clear that I don’t fully understand the structure of the repository, even though I’ve read the contributor guideline.

Could you please help me successfully close the pull request?

I have reformatted the files, and I’ve run the tests locally with no errors. It’s clear that I’m missing a setup or something else that I’m not aware of.

Thanks again for your support.

gberenice commented:
@carlolucadei thanks for the fixes!

The issue occurs because the test check starts two runs in parallel (one for Terraform, one for Opentofu). Some resources don’t have unique names, so the second job may fail due to resources not being cleaned up after the first job.
This requires some fixes to the tests, but it doesn’t block merging the PR since one of the test jobs was completed successfully.

Could you update your branch `KseniaSecurity:added-custom-name-to-capacity-provider` so I can merge it to base branch?

carlolucadei commented:
> @carlolucadei thanks for the fixes!
> 
> The issue occurs because the test check starts two runs in parallel (one for Terraform, one for Opentofu). Some resources don’t have unique names, so the second job may fail due to resources not being cleaned up after the first job. This requires some fixes to the tests, but it doesn’t block merging the PR since one of the test jobs was completed successfully.
> 
> Could you update your branch `KseniaSecurity:added-custom-name-to-capacity-provider` so I can merge it to base branch?

Hi @gberenice I'll do it asap. 
Meanwhile, our change is referred to that issue. We got the same error in the production environment and we started a new fork to extend the module.
In that way, we are able to specify the capacity provider name ;)

gberenice commented:
/terratest

gberenice commented:
@carlolucadei unfortunately there is a new issue with triggering the test job - some contributors do not have permission to run the tests. The CloudPosse team is already on it, and I'll get back to you as soon as the fix is released.

goruha commented:
/terratest

carlolucadei commented:
@gberenice I synced my fork, fixed the conflict locally in my repo, and updated the README.md by running:

```bash
make init
make readme
```
Then I pushed the changes again.
Let me know if it's okay, as in our pull request evaluation process, conflicts are typically handled by the reviewer.

gberenice reviewed (approved):
LGTM, thanks for working through this @carlolucadei!
